File: Sourcecode/FleetManager/BusinessLayer/models/Maintenance.cs
using BusinessLayer.models.general;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BusinessLayer.models
{
    public class Maintenance : GeneralModels
    {
        public Maintenance(DateTime date, double price, string garage)
        {
            Date = date;
            Price = price;
            Garage = garage;
        }
        public int Id { get; set; }

        public DateTime Date { get; set; }
        public double Price { get; set; }
        public string Garage { get; set; }
        public List<Invoice> Invoices { get; set; }
        public int RequestId { get; set; }
        public Request Request { get; set; }
    }
}

File: Sourcecode/front-end-angular/fleetmanager/src/app/models/maintenance.ts
export interface IMaintenance{
    date : Date;
    price : number;
    garage : string;
}

export class Maintenance implements IMaintenance{
    date: Date;
    price: number;
    garage: string;

    constructor(date : Date, price : number, garage : string){
        this.date = date;
        this.price = price;
        this.garage = garage;
    }

}